### PR TITLE
This fixes #18

### LINKED ISSUE #18: cdnify ignores paths in base URL
I'm having an issue where if my CDN URL contains a path like in the example (`/stuff/`) the URLs in the generated files do have the CDN URL prepended to them but they're missing the path. This is happening on images using root-relative paths.

My config looks like this:

```
cdnify: {
      dist: {
        options: {
          base: 'http://my.cdn.com/marketing/'
        },
        files: [{
          expand: true,
          cwd: '<%= app.dist %>',
          src: '**/*.{css,html,js}',
          dest: '<%= app.dist %>'
        }]
      }
    }
```

The generated URLs end up as `http://my.cdn.com/img/whatever`, missing the `/marketing` part. I made sure there were no other tasks doing work before or after CDNify that would interfere.

Right now I only notice it on images because they're the only thing referencing external assets in my project (I'm building a bunch of HTML emails).

Any ideas what could be wrong? For now my workaround is to use the rewriter function and match to any path where `indexOf('/img') === 0` but that won't work when assets aren't root-relative and its more than just images to worry about.

### PR DESCRIPTION
**Pull request by albogdano**
Just copied `joinBaseAndPath()` from 0.1.1 and replaced `url.resolve()` with it.

### CODE REVIEW

**Comment by XhmikosR** [comment]
We need a test case that this fixes and failed before. Otherwise I personally don't see the benefit of this.

**Comment by albogdano** [comment]
The benefit it that it doesn't discard some subdirs that are part of the base URL. For example `//cnd.com/subdir/12345/favicon.ico` where `12345` is discarded in current version. This was working fine in v0.1.1 and now it's broken.

**Comment by XhmikosR** [comment]
You will need to add a test then for this case.

**Comment by XhmikosR** [comment]
Let alone that this breaks Windows https://ci.appveyor.com/project/callumlocke/grunt-cdnify/build/27/job/paovun77cjx7j1n8.

**Comment by XhmikosR** [comment]
...which was the reason for using `url` in the first place than any custom code.

**Comment by albogdano** [comment]
I see. I'll try to come up with a better fix later.

File: tasks/cdnify.js [hunk 1]
```diff
@@ -73,7 +93,7 @@ module.exports = function (grunt) {
 
     if (typeof base === 'string') {
       rewriteURL = function (origUrl) {
-        return isLocalPath(origUrl) ? url.resolve(base, origUrl) : origUrl;
+        return isLocalPath(origUrl) ? joinBaseAndPath(options.base, origUrl) : origUrl;
```
Comment (XhmikosR) [inline tasks/cdnify.js]: `options.base` should be `base` since it's assigned to that 3 lines above.

**Comment by XhmikosR** [comment]
Can you paste a clear example of what you use and what you expect? See /tests/ folder.

Like

```
<link rel="apple-touch-icon" href="/apple-touch-icon.png">
```
```
<link rel="apple-touch-icon" href="//cdn.example.com/apple-touch-icon.png">
```

**Comment by albogdano** [comment]
Original line is something like:
`<link rel="apple-touch-icon" href="/images/apple-touch-icon.png">`

I'm using a CDN URL with subfolders in it, like `//cdn.example.com/static/201512241234/`.
So the expected result should be:
`<link rel="apple-touch-icon" href="//cdn.example.com/static/201512241234/apple-touch-icon.png">`

**Comment by XhmikosR** [comment]
OK, I can reproduce it now. I'll make a new branch with the failing test and see if I can fix it.

**Comment by albogdano** [comment]
Great, thanks a lot!

**Comment by XhmikosR** [comment]
`url.resolve` seems to do that. I tried using `url.parse` instead with no success so far. Any ideas are welcome, just make sure you use that branch so that we have a failing test case.

**Comment by albogdano** [comment]
It turns out that this is not standard behavior and should not be expected from this module unless there's a special option to allow it. I was also missing a `/` at the end of my CDN URL which was the main cause for my problems when upgrading to `1.x.x`. Please ignore this PR.

**Pull request closed without merging**